pyproject.toml

```diff
@@ -108,7 +109,7 @@ httpx = { extras = ["cli"], version = "*" }
 mypy = "^1.0.0"
 pytest-split = "^0.8.1"
 requests-mock = ">=1.11.0"
-types-requests = ">=2.31.0.2"
+types-requests = "==2.31.0.2"
```

Comment:
FYI: I did this downgrade to fix an issue where `poetry lock --no-update` hangs forever. We can't use later versions of `types-requests` because it depends on urllib > 2, which is not supported by boto.